src/main.cpp

```diff
@@ -30,19 +30,18 @@ unsigned long lastPublish = 0;
 
 // Publish a message
 void publish(String payload, DataQueue::PublishStatus status) {
+    if(status == DataQueue::DataBufferOverflow || dataQ.isCacheFull()) {
```

Comment:
I think to keep it consistent it makes sense to check for PublishStatus::PublisingAtMaxFrequency here and maybe have a separate check for whether or not the cache is full.  PublishingAtMaxFrequency warns us that the data queue is potentially going to fill up as opposed to telling us that it is full, so slightly different messages.  Unless you had a reason to get rid of it?

Reply:
Sorry no I didn't have a reason to get rid of it, I wanted to have consistent error behaviour and I got rid of it by accident. I've restored it now!